Code/Framework/AzCore/AzCore/Jobs/Job.h

```diff
@@ -234,319 +233,22 @@ namespace AZ
     {
         return m_context;
     }
 
-    AZ_FORCE_INLINE unsigned int Job::GetDependentCount() const
-    {
-        return (GetDependentCountAndFlags() & FLAG_DEPENDENTCOUNT_MASK);
-    }
-
-    AZ_FORCE_INLINE void Job::IncrementDependentCount()
-    {
-        AZ_Assert(GetDependentCount() < FLAG_DEPENDENTCOUNT_MASK, "Dependent count overflow");
-#ifdef AZCORE_JOBS_IMPL_SYNCHRONOUS
-        ++m_dependentCountAndFlags;
-#else
-        m_dependentCountAndFlags.fetch_add(1, AZStd::memory_order_acq_rel);
-#endif
-    }
-
-    inline void Job::IncrementDependentCountAndSetChildFlag()
-    {
-        AZ_Assert(GetDependentCount() < FLAG_DEPENDENTCOUNT_MASK, "Dependent count overflow");
-#ifdef AZCORE_JOBS_IMPL_SYNCHRONOUS
-        int oldCount = m_dependentCountAndFlags & FLAG_DEPENDENTCOUNT_MASK;
-        m_dependentCountAndFlags = (m_dependentCountAndFlags & ~FLAG_DEPENDENTCOUNT_MASK) | (oldCount + 1) | FLAG_CHILD_JOBS;
-#else
-        //use a single atomic operation to increment the count and set the child flag if possible
-        unsigned int oldCountAndFlags, newCountAndFlags;
-        do
-        {
-            oldCountAndFlags = m_dependentCountAndFlags.load(AZStd::memory_order_acquire);
-            int oldCount = oldCountAndFlags & FLAG_DEPENDENTCOUNT_MASK;
-            newCountAndFlags = (oldCountAndFlags & ~FLAG_DEPENDENTCOUNT_MASK) | (oldCount + 1) | FLAG_CHILD_JOBS;
-        } while (!m_dependentCountAndFlags.compare_exchange_weak(oldCountAndFlags, newCountAndFlags, AZStd::memory_order_acq_rel, AZStd::memory_order_acquire));
-#endif
-    }
-
-    inline void Job::DecrementDependentCount()
-    {
-#ifdef AZ_DEBUG_JOB_STATE
-        AZ_Assert((m_state == STATE_SETUP) || (m_state == STATE_STARTED)
-            || (m_state == STATE_PROCESSING) || (m_state == STATE_SUSPENDED), //child jobs
-            "Job dependent count should not be decremented after job is already pending");
-#endif
-        AZ_Assert(GetDependentCount() > 0, ("Job dependent count is already zero"));
-#ifdef AZCORE_JOBS_IMPL_SYNCHRONOUS
-        unsigned int countAndFlags = m_dependentCountAndFlags--;
-#else
-        unsigned int countAndFlags = m_dependentCountAndFlags.fetch_sub(1, AZStd::memory_order_acq_rel);
-#endif
-        unsigned int count = countAndFlags & FLAG_DEPENDENTCOUNT_MASK;
-        if (count == 1)
-        {
-            if (!(countAndFlags & FLAG_CHILD_JOBS))
-            {
-#ifdef AZ_DEBUG_JOB_STATE
-                AZ_Assert(m_state == STATE_STARTED, "Job has not been started but the dependent count is zero, must be a dependency error");
-                SetState(STATE_PENDING);
-#endif
-                m_context->GetJobManager().AddPendingJob(this);
-            }
-        }
-    }
-
-    inline AZ::s8 Job::GetPriority() const
-    {
-        return (GetDependentCountAndFlags() >> FLAG_PRIORITY_START_BIT) & 0xff;
-    }
-
 #ifdef AZ_DEBUG_JOB_STATE
-    AZ_FORCE_INLINE void Job::SetState(int state)
+    inline void Job::SetState(int state)
```

Comment:
This `inline` actually is already implicit (member function defined in a header) I believe (compared to a free function where the inline is necessary to avoid ODR violation)

Reply:
I believe it's only implicitly inline if the function body is inside the class definition body. Once you leave the class body you need to explicitly mark it inline.

Reply:
Yep that's true, I didn't notice this wasn't in the class declaration.